pkg/api/controller.go

```diff
@@ -1632,13 +1632,21 @@ func handleAPIError(w http.ResponseWriter, err error) bool {
 
 	case errors.Is(err, graveler.ErrLockNotAcquired):
 		writeError(w, http.StatusInternalServerError, "branch is currently locked, try again later")
+
 	case errors.Is(err, adapter.ErrDataNotFound):
 		writeError(w, http.StatusGone, "No data")
+
+	case errors.Is(err, db.ErrNotFound),
```

Comment:
1. This should probably be `auth.ErrNotFound` not `db.ErrNotFound` (currently they're the same, but they might not be...).
2. Why do we return http.StatusNotFound for other `ErrNotFound`s above?  More generally I don't think it will always be correct to return a 400 for all DB not founds.